packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -938,9 +891,19 @@ class HotRunner extends ResidentRunner {
           }
         } else {
           reassembleViews[view] = device.vmService;
-          reassembleFutures.add(device.vmService.flutterReassemble(
-            isolateId: view.uiIsolate.id,
-          ).catchError((dynamic error) {
+          // If the tool identified a change in a single widget, do a fast instead
```

Comment:
this method is pretty massive

Reply:
Yeah, historically it was really hard to change because we didn't have good test coverage. And the last time I changed it significantly I accidentally broke hot reload for some people :(